packages/SwingSet/src/kernel/kernel.js

```diff
@@ -286,35 +274,27 @@ export default function buildKernel(
     queueToKref(vatAdminRootKref, 'meterCrossedThreshold', args, 'logFailure');
   }
 
-  function deductMeter(meterID, compute, firstTime) {
-    assert.typeof(compute, 'bigint');
-    const res = kernelKeeper.deductMeter(meterID, compute);
-
-    // We record the deductMeter() in postAbortActions.meterDeductions. If
-    // the delivery is rewound for any reason (syscall error, res.underflow),
-    // then deliverAndLogToVat will repeat the deductMeter (which will repeat
-    // the notifyMeterThreshold), so their side-effects will survive the
-    // abortCrank(). But we don't record it (again) during the repeat, to
-    // make sure exactly one copy of the changes will be committed.
+  let terminationTrigger;
```

Comment:
It needs to be set by the two functions (`requestTermination` and `vatFatalSyscall`) which are handed to other code, so yeah I didn't see a way to avoid it.

Hm, it's worth considering whether we should move the flag into the vat-warehouse or the vat-manager. Basically after you tell vat-warehouse to make a delivery, you ask vat-warehouse whether the vat manager is now in a fatal-error state. Maybe the `KernelSyscallResult` that comes back to the vat-manager with a non `['ok', results]` status could trigger the manager to remember the problem, and then report it back to the delivery handler. That would help with our future plans for parallelism, for sure. I'll add a note.